### DISCUSSION TITLE
Contributing - DCO

### DISCUSSION THREAD

**Discussion by staticdev** [nextcloud/ansible-collection-nextcloud-admin#262] [Ideas]
@aalaesar @wiktor2200 it is kinda annoying to have broken PR because of DCO. Does some of you know a way to automate adding sign to all git commits? (also it would be great to add this to our contributing guide since we will receive a lot of PRs without signing and we could just point to a documentation helping them).

**Comment by wiktor2200** [comment]
Hi! We should definitely work on Contribution guide, as we started to get more and more pull requests.
DCO sign-off is quite annoying, because it cannot be added automatically via `git config` command (https://github.com/nextcloud/ansible-collection-nextcloud-admin/pull/236#issuecomment-1473649448), so the only option is to run `git commit -s` every time or at least create an alias for that in profile/bashrc or similar (depending on environment).

Or the other option (and probably the easiest) would be adding sign-off to `pre-commit` hook, so without DCO passed (if pre-commit installed) you won't be able to push. But I'm wondering whether this work, because push is done to fork (which we don't have access to), so maybe there is an option in repo settings to block commits without DCO.

**Reply by wiktor2200** [reply]
I think we could base our contribution guide on this manual: https://contribute.cncf.io/maintainers/templates/contributing
What is more we can create PR template with checklist and then we can add check for DCO, so this would avoid new PR being created without DCO.
